### ISSUE TITLE
Special case generated theme function for nested colors with "DEFAULT" key

### ISSUE THREAD

**Issue by nidico** [matheus23/elm-tailwind-modules#19]
Tailwind treats colors defined in color object syntax with `DEFAULT` key in a special way: The `DEFAULT` substring
 is skipped in the generated css definitions.

The following example from [the tailwind docs](https://tailwindcss.com/docs/customizing-colors#color-object-syntax), would create classes like `bg-tahiti` (not `bg-tahiti-DEFAULT`), `bg-tahiti-light`, and `bg-tahiti-dark`:

```js
module.exports = {
  theme: {
    colors: {
      // ...
      'tahiti': {
        light: '#67e8f9',
        DEFAULT: '#06b6d4',
        dark: '#0e7490',
      },
      // ...
    },
  },
}
```

As of elm-tailwind-modules 0.5.0-alpha1, a function `tahiti_DEFAULT` is generated in `Tailwind/Theme.elm` aside `tahiti_dark` and `tahiti_light`. For simplicity and consistency reasons, elm-tailwind-modules should do the same as tailwind and just call that function `tahiti`.

**Comment by matheus23** [comment]
Thanks for digging into this. This issue and #18 likely share the same fix. It should be fairly straight-forward, specifically, this is the code for expanding the recursive key-value record of colors into a flat key-value record:
https://github.com/matheus23/elm-tailwind-modules/blob/a6871fac5c506de38548cb978bfed537b01d452d/src/code-generators/tailwind-theme.ts#L128-L136

E.g. that code turns your example above into
```json
{
  "tahiti_light": "#67e8f9",
  "tahiti_DEFAULT": "#06b6d4",
  "tahiti_dark": "#0e7490",
}
```

If you'd be interested in contributing a fix, I'd be happy to merge a PR :)

In any case, let me know. I'd probably get to this this weekend otherwise.

**Comment by nidico** [comment]
> If you'd be interested in contributing a fix, I'd be happy to merge a PR :)

I shortly looked into it, but currently lack the time to properly dive into it.

**Comment by matheus23** [comment]
I've published a new version (0.5.0-alpha3), which should fix this issue. Let me know if it works on your end :)

**Comment by nidico** [comment]
Works perfectly!
